client/components/jetpack/card/jetpack-product-card/upgrade-nudge.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { translate as globalTranslate, useTranslate, TranslateResult } from 'i18n-calypso';
+import { isFinite } from 'lodash';
+import React, { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import Gridicon from 'components/gridicon';
+import { preventWidows } from 'lib/formatting';
+import PlanPrice from 'my-sites/plan-price';
+import { JETPACK_OFFER_RESET_UPGRADE_NUDGE_DISMISS } from 'my-sites/plans-v2/constants';
+import { savePreference } from 'state/preferences/actions';
+import { getPreference } from 'state/preferences/selectors';
+
+// This data should come from an API. It's now hardcoded until we figure out where is
+// this coming from.
+const DEFAULT_DESCRIPTION = globalTranslate(
+	'Additional security for sites with 24/7 activity.' +
+		' Recommended for eCommerce stores, news organizations, and online forums.'
+);
+
+const DEFAULT_FEATURES = [
+	{
+		title: globalTranslate( 'Real-Time Backup' ),
+		subtitle: globalTranslate( 'Every edit gets saved immediatelly' ),
+	},
+	{
+		title: globalTranslate( 'Real-Time Scan' ),
+		subtitle: globalTranslate( 'On-demand scanning' ),
+	},
+];
+
+type OwnProps = {
+	billingTimeFrame: TranslateResult;
+	className?: string;
+	currencyCode: string;
+	description?: TranslatedResult;
+	discountedPrice?: number;
+	features?: TranslatedResult[];
+	onUpgradeClick: () => void;
+	originalPrice: number;
+	productSlug: string;
+};
+
+const UpgradeNudge = ( {
+	billingTimeFrame,
+	className,
+	currencyCode,
+	description = DEFAULT_DESCRIPTION,
+	discountedPrice,
+	features = DEFAULT_FEATURES,
+	onUpgradeClick,
+	originalPrice,
+	productSlug,
+}: OwnProps ) => {
+	const translate = useTranslate();
+	const isDiscounted = isFinite( discountedPrice );
+
+	const storedPreference = useSelector( ( state ) =>
+		getPreference( state, JETPACK_OFFER_RESET_UPGRADE_NUDGE_DISMISS )
+	);
+
+	// Save dismiss to never show up the nudge again for this specific plan/product.
+	const dispatch = useDispatch();
+	const onCancelClick = useCallback( () => {
+		dispatch(
+			savePreference( JETPACK_OFFER_RESET_UPGRADE_NUDGE_DISMISS, {
+				...storedPreference,
+				[ productSlug ]: Date.now(),
+			} )
+		);
+	}, [ dispatch, productSlug, storedPreference ] );
+
+	// If this nudge was dismissed in the past don't show the nudge.
+	if ( storedPreference && storedPreference[ productSlug ] ) {
+		return null;
+	}
+
+	return (
+		<div className={ classNames( 'jetpack-product-card__nudge', className ) }>
+			<div className="jetpack-product-card__summary">
+				<div className="jetpack-product-card__nudge-heading">
+					<h3>{ translate( 'Upgrade to' ) }</h3>
+					<h2 className="jetpack-product-card__nudge-product-type">{ translate( 'Real-Time' ) }</h2>
```

Comment:
Semantically, the product type should probably be part of the `h3` heading.

Reply:
You're absolutely right!